### PR TITLE
feat: data augmentation enhancement

### PR DESCRIPTION
**Pull request by Owen-Liuyuxuan**
Here we included two important data augmentation enhancements.

1. Do not check augmentation validation on lane crossing but on road-border crossing. https://github.com/tier4/Diffusion-Planner/issues/364
2. Allow the augmentation not only started on t = 0, but also in t < 0s, so that the model will not only learn about creating a restoration curves in 2 seconds future, but also understand how to smoothly close from existing restoring paths.  -- This is somehow related to https://github.com/tier4/Diffusion-Planner/issues/362  (but not fully the same case because #362 is about bridge mode)
3. Instead of directly updating the current augmentation, I chose to implement a sub-class of the default optimization.

image

image

### CODE REVIEW

**Comment by Owen-Liuyuxuan** [comment]
model_1 default aug
model_2 tau aug

Test:
- Continue training the best model on a mini dataset `/mnt/storage_rdma/diffusion_planner/mini-datasets` that contains only one sequence. 
- the validation lateral position loss improved, and the ability to restoring from bias position improved
- Still we need larger scale testing

image

 
image

**Comment by Owen-Liuyuxuan** [comment]
Test on 20 epochs of SFT. -- Because of the poor setup, the SFT is not very perfect, but since it is not very perfect, we see from close loop performance that the ability to restoring back the path is improved --- this is exactly what I would love to see.

I think this level of tests are enough on dev-set. And as long as we keep quintic we will keep the original augmentation.


https://github.com/user-attachments/assets/34bc7cc0-8b5e-45e3-80f7-8ccc9175d641

image
From open loop result there are improvement on validation loss.
